### PR TITLE
Extend the maintainers documentation

### PR DESCRIPTION
**Pull request by andreeaflorescu**
Added some more details about what are the responsibilities of a repository maintainer and a gatekeeper.
Documented the process that we've been using for adding code owners.

### CODE REVIEW

**Review by gsserge — APPROVED** [review]

File: MAINTAINERS.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  are followed before publishing a new crate or a new version.
+- follow the [crate release process](docs/maintainers/crate_release.md) for
+  publishing new crate versions
+
+The repository maintainers are mentioned in the `CODEOWNERS` file of all
+rust-vmm repositories.
+
+### Becoming a Repository Maintainer
+
+1. To be a code owner, open a PR in the target repository, edit the `CODEOWNER`
```
Comment (lauralt) [inline MAINTAINERS.md:30]: CODEOWNER -> CODEOWNERS

File: MAINTAINERS.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+file and add your GitHub handle there. In case you want to add someone else as
+a code owner, open a PR in the target repository, add their GitHub handle in
+the `CODEOWNERS` file, and ask that person to formally approve the PR as
+acknowledgement that they agree with the role.
+2. Irrespective of who created the PR, all existing code owners (as defined in
+the`CODEOWNERS` file) must approve the PR before it is merged. In case this
```
Comment (lauralt) [inline MAINTAINERS.md:36]: `this` should be removed.

File: MAINTAINERS.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+there are no code owners in the repository, ask [2 gatekeepers](#gatekeepers)
+to approve the PR for you.
+3. Once the PR is merged, ask one of the existing code owners to grant you
```
Comment (lauralt) [inline MAINTAINERS.md:39]: Should we explain how this is done?

File: README.md [hunk 4]
```diff
@@ -207,6 +207,12 @@ All the rust-vmm repositories accept contributions by a
 [GitHub Pull Request (PR)](https://help.github.com/articles/using-pull-requests/).
 For more details, please check the [contributing document](CONTRIBUTING.md).
 
+For other processes (like setting up repos), please check the
+[`extended documentation](docs).
```
Comment (lauralt) [inline README.md:211]: `extended documentation -> extended documentation

**Comment by stsquad** [comment]
Looks good to me. I guess the CODEOWNERS vs MAINTAINERS thing is just the result of different community terminology. In QEMU for example we have MAINTAINERS and REVIEWERS where reviewers can provide review but maintainers still need to do the merge due to our tooling approach.

**Review by lauralt — APPROVED** [review]

**Review by sameo — APPROVED** [review]

**Pull request merged**